### PR TITLE
Remove golint issues

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Remove golint issues 
- Configure CI golint check
- Fix security issues

Report card: https://goreportcard.com/report/github.com/mszostok/capact

## Related issue(s)
- https://github.com/capactio/capact/issues/321

### CODE REVIEW

File: internal/cli/environment/delete/kind.go [hunk 1]
```diff
@@ -9,11 +9,13 @@ import (
 	"sigs.k8s.io/kind/pkg/cmd"
 )
 
+// KindOptions holds configuration for delete KIND cluster operation.
 type KindOptions struct {
 	Name       string
 	Kubeconfig string
 }
 
+// Kind delete a given KIND cluster.
```
Comment (pkosiec) [inline internal/cli/environment/delete/kind.go:18]: a) to be consistent with namings
b) kind is written simply as "kind" in its docs

```suggestion
// Kind delete a given kind cluster.
```

File: internal/cli/environment/delete/kind.go [hunk 2]
```diff
@@ -9,11 +9,13 @@ import (
 	"sigs.k8s.io/kind/pkg/cmd"
 )
 
+// KindOptions holds configuration for delete KIND cluster operation.
```
Comment (pkosiec) [inline internal/cli/environment/delete/kind.go:12]:
```suggestion
// KindOptions holds configuration for delete kind cluster operation.
```

File: pkg/sdk/dbpopulator/populator.go [hunk 3]
```diff
@@ -447,6 +447,7 @@ call apoc.periodic.iterate("MATCH (n:to_remove) return n", "DETACH DELETE n", {b
 yield batches, total return batches, total
 `
 
+// Populate neo4j database with Public Hub manifests.
```
Comment (pkosiec) [inline pkg/sdk/dbpopulator/populator.go:450]:
```suggestion
// Populate imports Public Hub manifests into a Neo4j database.
```

File: pkg/hub/api/graphql/public/helpers.go [hunk 4]
```diff
@@ -8,14 +8,18 @@ import (
...
 func NewActionImportsError(actionRef string) error {
 	return errors.Errorf("Full path not found in Implementation imports for action %q", actionRef)
 }
 
+// NewActionReferencePatternError returns an error indicating malformed Action import pattern path.
 func NewActionReferencePatternError(actionRef string) error {
 	return errors.Errorf("Action reference %q doesn't follow pattern <import_alias>.<method_name>", actionRef)
 }
 
+// ResolveActionPathFromImports resolve import alias and return full InterfaceReference used in Action.
```
Comment (pkosiec) [inline pkg/hub/api/graphql/public/helpers.go:21]:
```suggestion
// ResolveActionPathFromImports resolves import alias and returns full InterfaceReference used in Action.
```

File: internal/k8s-engine/validate/action.go [hunk 5]
```diff
@@ -11,16 +11,19 @@ import (
...
 }
 
+// NewActionValidator returns a new ActionValidator instance.
 func NewActionValidator(wfCli *wfclientset.Clientset) *ActionValidator {
 	return &ActionValidator{
 		wfCli: wfCli,
 	}
 }
 
+// Validate a given Action definition.
```
Comment (pkosiec) [inline internal/k8s-engine/validate/action.go:26]: Suggestion: While it is a valid sentence, I would recommend unifying that with the rest of the comments 🙂 For example with `// Update updates current Capact Policy configuration with a given input.`

```suggestion
// Validate validates a given Action definition.
```

File: internal/k8s-engine/policy/error.go [hunk 6]
```diff
@@ -2,4 +2,5 @@ package policy
 
 import "github.com/pkg/errors"
 
+// ErrPolicyConfigMapNotFound returns an error indicating that Policy cannot be found.
```
Comment (pkosiec) [inline internal/k8s-engine/policy/error.go:5]: Suggestion: 🤔 Maybe `is`/`defines` is better in this context?

```suggestion
// ErrPolicyConfigMapNotFound defines an error indicating that Policy cannot be found.
```

File: internal/k8s-engine/graphql/model/model.go [hunk 7]
```diff
@@ -67,7 +71,7 @@ func (f *ActionFilter) Match(item v1alpha1.Action) bool {
 	return true
 }
 
-// Input used for continuing Action rendering in advanced mode
+// AdvancedModeContinueRenderingInput used for continuing Action rendering in advanced mode
```
Comment (pkosiec) [inline internal/k8s-engine/graphql/model/model.go:74]:
```suggestion
// AdvancedModeContinueRenderingInput is used for continuing Action rendering in advanced mode.
```

File: internal/k8s-engine/graphql/model/model.go [hunk 8]
```diff
@@ -14,6 +14,7 @@ type ActionToCreateOrUpdate struct {
 	InputParamsSecret *v1.Secret
 }
 
+// SetNamespace set a given namespace to all required poperties.
```
Comment (pkosiec) [inline internal/k8s-engine/graphql/model/model.go:17]:
```suggestion
// SetNamespace sets a given namespace to all required poperties.
```

File: internal/k8s-engine/graphql/domain/policy/errors.go [hunk 9]
```diff
@@ -2,4 +2,5 @@ package policy
 
 import "github.com/pkg/errors"
 
+// ErrCannotConvertAdditionalInput returns an error indicating that converting additional input was not possible.
```
Comment (pkosiec) [inline internal/k8s-engine/graphql/domain/policy/errors.go:5]:
```suggestion
// ErrCannotConvertAdditionalInput defines an error indicating that converting additional input was not possible.
```

File: internal/k8s-engine/graphql/domain/action/service.go [hunk 10]
```diff
@@ -34,6 +36,7 @@ func NewService(log *zap.Logger, actionCli client.Client) *Service {
 // Validate the list of input TypeInstances with validation webhook,
 // to make sure there are no TypeInstances with duplicated names and different IDs
 
+// Create Action on cluster side in the Namespace extracted from a given ctx.
```
Comment (pkosiec) [inline internal/k8s-engine/graphql/domain/action/service.go:39]:
```suggestion
// Create creates Action on cluster side in the Namespace extracted from a given ctx.
```

File: internal/k8s-engine/graphql/domain/action/service.go [hunk 11]
```diff
@@ -60,6 +63,7 @@ func (s *Service) Create(ctx context.Context, item model.ActionToCreateOrUpdate)
 	return item.Action, nil
 }
 
+// Update Action on cluster side in the Namespace extracted from a given ctx.
```
Comment (pkosiec) [inline internal/k8s-engine/graphql/domain/action/service.go:66]:
```suggestion
// Update updates Action on cluster side in the Namespace extracted from a given ctx.
```

File: internal/getter/getter.go [hunk 12]
```diff
@@ -8,10 +8,11 @@ import (
 	"github.com/pkg/errors"
 )
 
+// Download data from a given source to local file system under a given destination path.
```
Comment (pkosiec) [inline internal/getter/getter.go:11]: Suggestion:
```suggestion
// Download downloads data from a given source to local file system under a given destination path.
```

File: internal/cli/upgrade/upgrade.go [hunk 13]
```diff
@@ -95,6 +92,7 @@ func New(w io.Writer) (*Upgrade, error) {
 	}, nil
 }
 
+// Run Capact cluster upgrade action with a input configuration.
```
Comment (pkosiec) [inline internal/cli/upgrade/upgrade.go:95]:
```suggestion
// Run executes Capact cluster upgrade action with a input configuration.
```

File: internal/cli/terminal.go [hunk 14]
```diff
@@ -1,4 +1,4 @@
-// Copied and adjusted code from: https://github.com/kubernetes-sigs/kind/blob/b6bc112522651d98c81823df56b7afa511459a3b/pkg/internal/env/term.go#L51
+// Package cli is a copied and adjusted code from: https://github.com/kubernetes-sigs/kind/blob/b6bc112522651d98c81823df56b7afa511459a3b/pkg/internal/env/term.go#L51
```
Comment (pkosiec) [inline internal/cli/terminal.go:1]:
```suggestion
// Package cli is copied and adjusted code from: https://github.com/kubernetes-sigs/kind/blob/b6bc112522651d98c81823df56b7afa511459a3b/pkg/internal/env/term.go#L51
```

File: internal/cli/printer/static_spinner.go [hunk 15]
```diff
@@ -11,20 +11,24 @@ type StaticSpinner struct {
...
 	s.active = true
 	fmt.Fprintf(s.w, " • %s\n", stage)
 }
 
+// Active returns whether or not the spinner is currently active.
 func (s *StaticSpinner) Active() bool {
 	return s.active
 }
 
+// Stop the spinner with a given message.
```
Comment (pkosiec) [inline internal/cli/printer/static_spinner.go:30]: Suggestion:
```suggestion
// Stop deactivates the spinner with a given message.
```

File: internal/cli/printer/static_spinner.go [hunk 16]
```diff
@@ -11,20 +11,24 @@ type StaticSpinner struct {
 	active bool
 }
 
+// NewStaticSpinner returns a new StaticSpinner instance.
+func NewStaticSpinner(w io.Writer) *StaticSpinner {
+	return &StaticSpinner{w: w}
+}
+
+// Start the spinner with a given name.
```
Comment (pkosiec) [inline internal/cli/printer/static_spinner.go:19]:
```suggestion
// Start activates the spinner with a given name.
```

File: internal/cli/printer/resource.go [hunk 17]
```diff
@@ -36,40 +39,48 @@ func NewForResource(w io.Writer, opts ...ResourcePrinterOption) *ResourcePrinter
...
 func (r *ResourcePrinter) RegisterFlags(flags *pflag.FlagSet) {
 	flags.VarP(&r.output, "output", "o", fmt.Sprintf("Output format. One of: %s", r.availablePrinters()))
 }
 
+// PrintFormat returns default print format type.
 func (r *ResourcePrinter) PrintFormat() PrintFormat {
 	return r.output
 }
 
+// Print received object in requested format.
```
Comment (pkosiec) [inline internal/cli/printer/resource.go:83]: Suggestion:
```suggestion
// Print prints received object in requested format.
```

File: internal/cli/printer/dynamic_spinner.go [hunk 18]
```diff
@@ -13,22 +13,26 @@ type DynamicSpinner struct {
...
 	d.underlying.Suffix = fmt.Sprintf(" %s", stage)
 	d.underlying.Start()
 }
 
+// Active returns whether or not the spinner is currently active.
 func (d *DynamicSpinner) Active() bool {
 	return d.underlying.Active()
 }
 
+// Stop the spinner with a given message.
```
Comment (pkosiec) [inline internal/cli/printer/dynamic_spinner.go:35]:
```suggestion
// Stop deactivates the spinner with a given message.
```

File: internal/cli/printer/dynamic_spinner.go [hunk 19]
```diff
@@ -13,22 +13,26 @@ type DynamicSpinner struct {
...
 }
 
+// NewDynamicSpinner returns a new DynamicSpinner instance.
 func NewDynamicSpinner(w io.Writer) *DynamicSpinner {
 	return &DynamicSpinner{
 		underlying: spinner.New(spinner.CharSets[11], 100*time.Millisecond, spinner.WithWriter(w)),
 	}
 }
 
+// Start the spinner with a given name.
```
Comment (pkosiec) [inline internal/cli/printer/dynamic_spinner.go:23]:
```suggestion
// Start activates the spinner with a given name.
```

**Review by pkosiec — COMMENTED** [review]
Looks good, just few small improvements 🙂

File: internal/k8s-engine/policy/error.go (hunk 6 above)
Comment (mszostok) [reply internal/k8s-engine/policy/error.go:5]: yep, let's stay with `defines` as it is used already in other places

File: .golangci.yml [hunk 20]
```diff
@@ -13,11 +11,28 @@ issues:
...
+    - EXC0006
+    - EXC0007
+    - EXC0008
+    - EXC0009
+    - EXC0010
+    - EXC0011
+    - EXC0012
+    - EXC0013
+    - EXC0014
+    - EXC0015
```
[the remark this answers is not in the document]
Comment (mszostok) [reply .golangci.yml:30]: yep, good suggestion. I will also add a reason why I had to do it.

[the remark this answers is not in the document]
Comment (mszostok) [reply .golangci.yml:30]: EDIT :D 
I removed those exclude rules, as I fixed all of them without knowing that. The problem was that I should had
```
  exclude:
    # EXC0001 errcheck: Almost all programs ignore errors on these functions and in most cases it's ok
    - Error return value of .((os\.)?std(out|err)\..*|.*Close|.*Flush|os\.Remove(All)?|.*print(f|ln)?|os\.(Un)?Setenv). is not checked
```

and not
```
  exclude:
    - EXC0001
```

**Review by pkosiec — APPROVED** [review]
🚀

**Pull request merged**